File: workers/worker-v8/src/worker_runner.js
import { Redis } from "ioredis";
import { registerEngines } from "./registry.js";
import { listenAndProcess } from "./processor.js";

async function main() {
  const redisUrl = process.env.REDIS_URL || "redis://redis:6379";
  const redisClient = new Redis(redisUrl);
  const pubClient = new Redis(redisUrl);

  await registerEngines(redisClient);
  await listenAndProcess(redisClient, pubClient);
}

main().catch(console.error);

File: workers/worker-c-cpp/src/worker_runner.py
import os
import redis
from worker_c_cpp.registry import register_engines
from worker_c_cpp.processor import listen_and_process


def main():
    redis_url = os.environ.get("REDIS_URL", "redis://redis:6379")
    redis_client = redis.Redis.from_url(redis_url, decode_responses=True)

    register_engines(redis_client)
    listen_and_process(redis_client)


if __name__ == "__main__":
    main()